### PR TITLE
Dt1 editor

### LINKED ISSUE #77: Add Editor Features to DT1 viewer
Depends on:
* https://github.com/OpenDiablo2/OpenDiablo2/issues/909
* #67

### PR DESCRIPTION
**Pull request by gucio321**
fixes #77

### CODE REVIEW

File: hscommon/hsenum/tile.go [hunk 1]
```diff
@@ -23,3 +23,40 @@ const (
 	TileLowerWallsEquivalentToRightLeftNorthCornerWall
 	TileLowerWallsEquivalentToSouthCornerwall
 )
+
+// GetTileTypeString returns string of tile type
+// nolint:gocyclo // can't reduce
+func GetTileTypeString(t int32) string {
```
Comment (gravestench) [inline hscommon/hsenum/tile.go]: we could probably do `type tileType int32` and implement the stringer interface. then, this func would become a method `String() string` of the `tileType` type. What is currently here is fine, but implementing the stringer interface is probably less taboo.

Also, the reason for the nolint directive is not quite right, we dont even need a nolint directive if a `map[int32]string` were declared inside of the func and you just returned `stringMap[t]`

a combination of both points above:
```golang
type tileType int32

func (t tileType) String() string {
	strings := map[tileType]string{
		TileFloor: "floor",
		TileSpecialTile1: "special",
		TileSpecialTile2: "special",
		TileShadow: "shadow",
		TileTree: "wall/object",
		TileRoof: "roof",
		TileLeftWall: "Left Wall",
		TileRightWall: "Upper Wall",
		TileRightPartOfNorthCornerWall: "Upper part of an Upper-Left corner",
		TileLeftPartOfNorthCornerWall: "Left part of an Upper-Left corner",
		TileLeftEndWall: "Upper-Right corner",
		TileRightEndWall: "Lower-Left corner",
		TileSouthCornerWall: "Lower-Right corner",
		TileLeftWallWithDoor: "Left Wall with Door object, but not always",
		TileRightWallWithDoor: "Upper Wall with Door object, but not always",
	}

	str, found := strings[t]

	if !found {
		str = "lower wall ?"
	}

	return str
}
```

Comment (gucio321) [reply hscommon/hsenum/tile.go]: IMO it should be method of `d2enum.TileType`

Comment (gravestench) [reply hscommon/hsenum/tile.go]: Agreed

Comment (gucio321) [reply hscommon/hsenum/tile.go]: OpenDiablo2/OpenDiablo2#1089

File: hswidget/dt1widget/helpers.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	case d2enum.TileSouthCornerWall:
+		return "Lower-Right corner"
+	case d2enum.TileLeftWallWithDoor:
+		return "Left Wall with Door object, but not always"
+	case d2enum.TileRightWallWithDoor:
+		return "Upper Wall with Door object, but not always"
+	default:
+		return "lower wall ?"
+	}
+}
```
Comment (gravestench) [inline hswidget/dt1widget/helpers.go]: https://github.com/OpenDiablo2/OpenDiablo2/pull/1089 has been merged, you can change to the method of d2enum.TileType

Comment (gucio321) [reply hswidget/dt1widget/helpers.go]: 0b7c808

File: hswidget/dt1widget/helpers.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+func decodeTileGfxData(blocks []d2dt1.Block, floorPixBuf, wallPixBuf *[]byte, tileYOffset, tileWidth int32) {
+	for i := range blocks {
+		switch blocks[i].Format() {
+		case d2dt1.BlockFormatIsometric:
+			decodeFloorBlock(&blocks[i], floorPixBuf, tileYOffset, tileWidth)
+		case d2dt1.BlockFormatRLE:
+			decodeWallBlock(&blocks[i], wallPixBuf, tileYOffset, tileWidth)
+		}
+	}
+}
```
Comment (gravestench) [inline hswidget/dt1widget/helpers.go]: should we just export some of these funcs from od2?

Comment (gucio321) [reply hswidget/dt1widget/helpers.go]: e4ea3dc

**Review by essial — APPROVED** [review]

**Pull request merged**